src/ucp/wireup/select.c

```diff
@@ -1426,29 +1427,94 @@ ucp_wireup_is_md_map_count_valid(ucp_context_h context, ucp_md_map_t md_map)
            (ucs_popcount(md_map) < UCP_MAX_OP_MDS);
 }
 
+static double ucp_wireup_get_lane_progress_overhead(ucp_worker_h worker,
+                                                    ucp_rsc_index_t rsc_index)
+{
+    ucp_worker_iface_t *wiface = ucp_worker_iface(worker, rsc_index);
+    uct_perf_attr_t perf_attr;
+
+    perf_attr.field_mask = UCT_PERF_ATTR_FIELD_OPERATION |
+                           UCT_PERF_ATTR_FIELD_PROGRESS_OVERHEAD;
+
+    /* We use AM_SHORT as a dummy operation because we don't know which
+        protocol will be selected (anyway it has no effect on the value of
+        progress_overhead). */
+    perf_attr.operation = UCT_EP_OP_AM_SHORT;
+
+    if (uct_iface_estimate_perf(wiface->iface, &perf_attr) != UCS_OK) {
```

Comment:
why can't we rely on just BW? We already filtering lanes  according to the BW, but on a later stage, when RMA BW lanes already setup. Maybe we can move this filtering earlier (here), so slow lanes will not be initialized at all?

Reply:
BW filtering cause some issue, because it also filters IB transport from intra node endpoints (where you have SHM transports).

Reply:
that's an option but wouldn't it be a tailor made solution only for this specific issue?